### PR TITLE
Add zustand and redux-toolkit to the demo.

### PR DESCRIPTION
**Pull request by MortezaMirjavadi**
Add example to the demo for zustand and redux-toolkit. These libraries are powerful state management. Zustand in actually is a micro state management and work by hooks, also even use as a global state management.

### CODE REVIEW

File: demo/package-lock.json [hunk 1]
```diff
@@ -8172,6 +19514,15 @@
       "integrity": "sha1-1cdSgl5TZ+eG944Y5EXqIjoVWVI=",
       "dev": true
     },
+    "string_decoder": {
```
Comment (JoviDeCroock) [inline demo/package-lock.json:19517]: this change seems to reorder a lot of stuff in here, would it be possible to reduce the amount of change in the lockfile? :sweat_smile:

**Review by JoviDeCroock — APPROVED** [review]

File: demo/package-lock.json (hunk 1 above)
Comment (MortezaMirjavadi) [reply demo/package-lock.json:19533]: It's done.

Comment (JoviDeCroock) [reply demo/package-lock.json:19533]: still sais 11500 changes to the lockfile for me

Comment (JoviDeCroock) [reply demo/package-lock.json:19533]: It changes the version of every package, including those of our root deps that can't be the intention 😅 you are adding things to _demo_ and things are changing in the root folder

Comment (MortezaMirjavadi) [reply demo/package-lock.json:19533]: Got it, please check again. @JoviDeCroock

Comment (MortezaMirjavadi) [reply demo/package-lock.json:19533]: > It changes the version of every package, including those of our root deps that can't be the intention 😅 you are adding things to _demo_ and things are changing in the root folder

Done.

**Review by rschristian — APPROVED** [review]
Sorry for the long delay here, just slipped through.

I had to rewrite the commit(s) as the Git state was absolutely borked, must've been a bad auto-merge from the GitHub UI.

**Pull request merged**